### PR TITLE
Petting is better

### PR DESCRIPTION
**Pull request by TrustyGun**
The heart animation that happens when you pet dog and cats has been changed to use an overlay and thus it's smoother

<3

### CODE REVIEW

File: code/game/objects/effects/overlays.dm [hunk 1]
```diff
@@ -619,6 +619,17 @@
...
+/obj/effect/overlay/temp/heart
+	name = "heart"
+	icon = 'icons/mob/animal.dmi'
+	icon_state = "heart"
+	duration = 25
+
+/obj/effect/overlay/temp/heart/Initialize(mapload)
+	pixel_x = rand(-4,4)
+	pixel_y = rand(-4,4)
+
```
Comment (ChangelingRain) [inline code/game/objects/effects/overlays.dm]: . = ..() must be called in Initialize

Comment (TrustyGun) [inline code/game/objects/effects/overlays.dm]: fixed

**Comment by kevinz000** [comment]
why is this not headpat animation..?

**Review by Cyberboss — APPROVED** [review]
Doesn't need a changelog for a code refactor

**Pull request merged**